compiler/resolution/resolveFunction.cpp

```diff
@@ -143,6 +149,91 @@ static void resolveFormals(FnSymbol* fn) {
       if ((useIntent & INTENT_FLAG_IN))
         formal->intent = useIntent;
     }
+
+    if (formal->defaultExpr != NULL && fn->hasFlag(FLAG_EXPORT)) {
+      storeDefaultValuesForPython(fn, formal);
+    }
+  }
+}
+
+// When compiling for Python interoperability, default values for arguments
+// should get propagated to the generated Python files.
+static void storeDefaultValuesForPython(FnSymbol* fn, ArgSymbol* formal) {
+  if (fLibraryPython) {
+    Expr* end = formal->defaultExpr->body.tail;
+
+    if (SymExpr* sym = toSymExpr(end)) {
+      VarSymbol* var = toVarSymbol(sym->symbol());
+
+      // Might be an ArgSymbol instead of a VarSymbol
+      if (var && var->isImmediate()) {
+        Immediate* imm = var->immediate;
+        std::stringstream ss;
```

Comment:
I find it a bit odd that you needed a std::stringstream as it's just converting int to string etc... I'd think there'd be a better method to convert an immediate to a string. Could you make a GenRet from the immediate? Or do immediates have some other method? Anyway this code should probably be factored out into an immediate support file.